rfcs/0018-bfd-release-versioning.md

```diff
@@ -0,0 +1,339 @@
+- Define a Release Management Process: this is just the first step in pursuing some wider release management process for BFD. Some release management scenarios have been used here to describe what's possible once we have a consistent versioning strategy in place, but this RFC does not seek to define any process beyond versions and component version resolution.
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+BFD uses a version string that shares the _format_ described by [semver.org@2.0.0](https://semver.org/spec/v2.0.0.html#summary) with practical deviations from the specification.
+Each component found in the familiar `MAJOR.MINOR.PATCH` format will be further described in the [detailed design below](#proposed-solution-detailed-design), but the modified algorithm for calculating the components can be summarized as follows:
+
+```
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+MAJOR version when you add a database schema migration
+MINOR version when you add non-database schema migration changes to the application source code
+PATCH version when you add non-application source code changes
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+```
+> NOTE: "application source code" above refers to the source code directly responsible for the resultant Java Archive (JAR) artifacts
+
+When applied, the version string clarifies release artifacts. As of March 2023, these artifacts should include:
+- [GitHub Release](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases)...
+  - [Git Source Code Tag Reference](https://git-scm.com/book/en/v2/Git-Internals-Git-References#_tags)
+  - Release Notes
+  - Change Log Documentation
+  - Data Dictionary Artifacts
+  - OpenAPI Artifacts
+- [Amazon Machine Images (AMI)](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html) as [technical resource tags](https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html#tag-categories)... 
+  - `bfd-db-migrator`
+  - `bfd-pipeline`
+  - `bfd-server`
+  - `bfd-server-load`
+- [Container Images](https://aws.amazon.com/containers/) as [container image tags](https://docs.docker.com/engine/reference/commandline/image_tag/)...
+  - `bfd-server-load-node`
+  - `bfd-server-regression`
+  - `bfd-synthea-generation`
+
+### Proposed Solution: Detailed Design
+[Proposed Solution: Detailed Design]: #proposed-solution-detailed-design
+
+Deviations from a standard must be carefully considered before they are accepted by anyone.
+What follows are some justifications for the modified semver.org@2.0.0 component version algorithms that are consistent with the BFD system.
+
+#### Incrementing `MAJOR` Component Version
+
+```
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+MAJOR version when you add a database schema migration
```

Comment:
Isn't this one a bit drastic? For the amount of db migrations, we'd be versioning in the (potentially) tens if not hundreds.

Reply:
Happily, you're on the _right_ side of this one already–0018 is the versioning RFC we've decided to reject, whereas the versioning RFC 0019 is just going to be some vanilla semantic versioning based on the latest GA'd version of the BFD FHIR API server.